includes/core/server/request/class-update-account.php

```diff
@@ -0,0 +1,234 @@
+<?php
+/**
+ * Class file for WCPay\Core\Server\Request\Update_Account.
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core\Server\Request;
+
+use WCPay\Core\Exceptions\Server\Request\Invalid_Request_Parameter_Exception;
+use WCPay\Core\Server\Request;
+use WC_Payments_API_Client;
+
+/**
+ * Request class for updating account.
+ */
+class Update_Account extends Request {
+	/**
+	 * Returns the request's API.
+	 *
+	 * @return string
+	 */
+	public function get_api(): string {
+		return WC_Payments_API_Client::ACCOUNTS_API;
+	}
+
+	/**
+	 * Returns the request's HTTP method.
+	 */
+	public function get_method(): string {
+		return 'POST';
+	}
+
+	/**
+	 * If true, the request will be signed with the user token rather than blog token.
+	 *
+	 * @return bool
+	 */
+	public function should_use_user_token(): bool {
+		return true;
+	}
+
+	/**
+	 * Used to prepare request from array account_settings.
+	 *
+	 * @param  array $account_settings  Account settings: key is the param name, value is the param value.
+	 *
+	 * @return static
+	 * @throws Invalid_Request_Parameter_Exception When either no account settings provided or no existing setter for provided parameter.
+	 */
+	public static function from_account_settings( array $account_settings ) {
+		if ( 0 === count( $account_settings ) ) {
+			throw new Invalid_Request_Parameter_Exception(
+				__( 'No account settings provided', 'woocommerce-payments' ),
+				'wcpay_core_invalid_request_parameter_account_settings_empty'
+			);
+		}
+
+		$wcpay_request = static::create();
+
+		foreach ( $account_settings as $param_name => $value ) {
+			$param_setter = 'set_' . $param_name;
+
+			if ( ! method_exists( static::class, $param_setter ) ) {
```

Comment:
I wouldn't throw an exception here. I would simply skip it and continue with the loop. In case someone ads something to the account settings, this API endpoint would crash.

Reply:
Make sense, addressed bcf2bc6d20bbfc277df710755a7dd8193084d162